### PR TITLE
Add Redux in React component

### PR DESCRIPTION
**Pull request by Metaverse-Nabeel**
### The following changes have been made to this PR:

- Wrapped your <App> with the <Provider> component from react-redux
- Passed configured store into the <Provider>
- Created an initialState variable for booksSlice,
- Displayed books, received from the slice, in a reusable component
- Dispatched actions using useDispatch
- Added a <Button> component
- An event handler that adds a book to the books array (with attributes id, title and author)
- Added a <Button> component;
- An event handler that removes a book from the books array (by id)
- Corrected Linter Errors

### CODE REVIEW

File: src/components/Book.jsx [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
-    {
-      id: 3,
-      title: 'Edit',
-    },
-  ];
+  const dispatch = useDispatch();
+
+  const handleRemove = (bookId) => {
+    dispatch(removeBook(bookId));
+  };
```
Comment (pinkmoon25) [inline src/components/Book.jsx:15]: Good job so far :clap: Just one issue we can fix, please see that the remove action is not working, if i click on remove button nothing seems to happen, please fix this

[Screencast from 12-05-23 09:35:38 PM IST.webm](https://github.com/Metaverse-Nabeel/BookStore/assets/97521557/052af21c-9067-4293-9078-afdbbf38be4f)

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @Metaverse-Nabeel , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Linters :heavy_check_mark: 
Gitflow :heavy_check_mark: 
Add Action :heavy_check_mark:
Redux store :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.jsx (hunk 1 above)
Comment (Metaverse-Nabeel) [reply src/components/Book.jsx:15]: Hello @PinkMoon25 

Thank you so much for your kind feedback. I shall resolve the issue and will again ask for a review.

Regards

**Review by danmainah — APPROVED** [review]
Hi @Metaverse-Nabeel , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Github flow was correctly used
✔️  All functionalities added successfully

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**